cis194/week7/stevemao/Scrabble.hs

```diff
@@ -0,0 +1,68 @@
+{-# OPTIONS_GHC -Wall #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving, FlexibleInstances #-}
+module Scrabble where
+import Data.Char
+import Buffer
+import JoinList
+import Sized
+import Editor
+
+newtype Score = Score Int
+  deriving (Eq, Ord, Show, Num)
+
+instance Semigroup Score where
+  (<>) = (+)
+
+instance Monoid Score where
+  mempty = Score 0
+
+score :: Char -> Score
+score a
+  | toUpper a `elem` ['A', 'E', 'I', 'L', 'N', 'O', 'R', 'S', 'T', 'U'] = Score 1
+  | toUpper a `elem` ['D', 'G'] = Score 2
+  | toUpper a `elem` ['B', 'C', 'M', 'P'] = Score 3
+  | toUpper a `elem` ['F', 'H', 'V', 'W', 'Y'] = Score 4
+  | toUpper a == 'K' = Score 5
+  | toUpper a `elem` ['J', 'X'] = Score 8
+  | toUpper a `elem` ['Q', 'Z'] = Score 10
+score _ = Score 0
+
+scoreString :: String -> Score
+scoreString = sum . fmap score
```

Comment:
Thought exercise:
Is it the same to write `scoreString = foldMap score`? and why

Reply:
Yeah, it maps each element of the structure to a monoid, and combine (`mappend`) the results.